### ISSUE TITLE
Regret moving Collectfast to jazzband

### ISSUE THREAD

**Issue by antonagestam** [jazzband/roadies#72]
Hi roadies,

I've been slow on finalizing moving Collectfast to Jazzband. Decoupling from integration tests talking to my personal s3 bucket turned out to not be straight forward. Today a Jazzband member merged a pull request that I had been holding off on merging, merging in commits that I had intended to squash into one and that would have been fixed in another pull request that fixed the issue in a better way. https://github.com/jazzband/collectfast/pull/99

The same person then pushed commits directly to the master branch, without creating pull requests, which for me is a huge no, especially in a collaborative setting such as Jazzband.

This has made me regret moving Collectfast to Jazzband. I thought there would be methods in place for reaching a consensus of what to merge into a project and for how collaboration should take place and was under the impression I would still be involved in decisions regarding the way forward for the project.

I especially think it's really important for guidelines to be in place for how to manage code review. 

Not sure where to go from here, but I kinda feel like I want the repository transferred back to my account. Please share your thoughts on the issue.

**Comment by blueyed** [comment]
Maybe https://help.github.com/articles/about-codeowners/ could be used, to require approval from a code owner?

/cc @Flimm

**Comment by Flimm** [comment]
@antonagestam That person was me. I didn't see the other pull request, I'm sorry about that. I hope my contributions improved the code-base, even if they didn't go the better route that you described.

Obviously, Jazzband members are free to create pull requests if they would like a second pair of eyes to review the changes, but I was under the impression that the goal of Jazzband was to make that optional rather than mandatory. If you would like to make that mandatory again, that makes sense to me, and in that case, let's move collectfast out of Jazzband and back to your account.

> Not sure where to go from here, but I kinda feel like I want the repository transferred back to my account. Please share your thoughts on the issue.

That makes sense to me.

**Comment by antonagestam** [comment]
@blueyed That won't prevent members from comitting to master, right?

@Flimm Just to make it super clear, my problem here is not with your contributions specifically, but rather the lack of processes/guidelines.

**Comment by antonagestam** [comment]
Before moving the project to Jazzband, open discussions about new features happened in Github Issues where they were vetted and improved before they were merged. The final decision always landed on me. In the context of Jazzband, I don't think that should land on one person specifically since that would defeat some of its purpose, but I think there needs to be some process in place for decision making, code reviews and feature vetting.

If there is no process in place, nothing prevents me as a member from reverting the commits of another member and implementing a feature in "my way". I think the goal should be to have guidelines that help us find "our way" together instead.

**Comment by triat** [comment]
Yea I kind of had the same feeling when I joined Jazzband, there is not special rule on who can push what ... I would have expected a small check on the capacities of the developers to make sure that we don't ends up with broken repo because people mess up with it. 

I don't know the position of the rookies on that, I just have the feeling that having different kind of developers might help to avoid this kind of "problem"

**Comment by antonagestam** [comment]
It also doesn't help that I don't have admin rights for the project anymore, so I can't make master a protected branch.

I've added some guidelines in the PR referenced above, would like to get your thoughts on its contents.

**Comment by camilonova** [comment]
I would like to add that moving somehow is better than not moving at all.

One of the reasons a lot of packages are not moving is because process is complicated or they are depending only on one person. We as a community must push towards giving others support to take their own choices and decisions in favor of moving the project forward, otherwise they will stay the same way as it is, and thats is the main reason we see a lot of abandoned projects.

@antonagestam I get your pain, is not easy to let go, but moving a project into jazzband means to me the original author does not have enough time to keep with the development phase and leaves the community a choice to work/improve the library any way they can.

I don't meant to get into a debate. I just remembered that I made "mistakes" too, just like @Flimm , but some contributions are better than none contributions.

I'm sure we all go with the best intentions on making our libraries better.

**Comment by mbeacom** [comment]
How is this repository proceeding? It would probably be ideal to allow original repository owners to establish merge process at their discretion. I sympathize with @antonagestam, but understand the need to keep open source repositories alive.

While I understand that projects migrated to jazzband can benefit from loose code merging for speed, I believe that code reviews are an essential step in not only developing quality software, but growing as a developer. Syntax and similar critiques are one thing, as we all have different standards and styles for how we write code, but we all make mistakes.

Merging directly to master is bad practice, no matter how to slice it.

Additionally, What's to stop someone from refactoring an entire project because they prefer one standard over another?

I feel like there needs to be some form of control over the consistency and flow of any given repository, otherwise it could just end in "organized" chaos.

Encouraging well thought out, quality commits that could be scrutinized makes everyone better. I'm not sure how you're supposed to control proper semantic versioning if you have random code being merged directly into master without any release notes or documentation.  This could ultimately lead to a dead repository versus a stale one.

In any case, I empathize with @Flimm. Don't beat yourself up about it.

I believe jazzband is a great idea, but some maturity of process might be advantageous and beneficial for the community as a whole.

**Comment by antonagestam** [comment]
@jezdez At this point I think the best thing for the project would be if you would transfer it back to my account so that it can move forward.

Please tell me if there is anything I can do to help you decide on this issue. If you think improved guidelines is a good idea, please let us know and I am sure there are plenty of members willing to help authoring them.

**Comment by mbeacom** [comment]
It's kind of sad to see a migration to jazzband leading to a dead repository versus a more active one.
In any case, hopefully this gets cleared up as it seems to have caused a deadlock on PRs being merged to `collectfast`.

Is there anything that can be done to get the ball rolling again? Would you be open to new guidelines @antonagestam or is @jezdez transferring the repository back the only resolution left at this point?

**Comment by antonagestam** [comment]
@jezdez Please move the project back to my account so that I don't have to fork the repository and lose the issues history.

**Comment by jezdez** [comment]
@antonagestam Sorry for the slow response folks, moving this now.

**Comment by jezdez** [comment]
@antonagestam For the record, GitHub doesn't support moving a repo from an organization to an account, so I moved it first to my account and then requested the move to your account from there.

**Comment by tony** [comment]
One thing I just bumped into is, for some tweaks, I do *sometimes* go right into master (stuff for documentation, typos, whatever). When those tiny things mess up, I noticed there's no ability to ``--amend`` + ``--force`` it away. Yeah, I have a lot of experience, I make mistakes, and within seconds I've done amend+force. Not proud of it, strive to be better, but it happens. Heh

Pushing to master gets to be a problem is when people do changes to the python code specifically. That *has* broken stuff for me on other django projects. And they were good programmers + had experience with django packages.

There's strict dogma of making a PR for *every* change, you could take it to the point where things can't merge until tests greenlight. But for a typo? There's also tradeoffs where red tape makes no sense.
